openedx_tagging/core/tagging/rest_api/v1/views.py

```diff
@@ -239,6 +264,57 @@ def export(self, request, **_kwargs) -> HttpResponse:
 
         return HttpResponse(tags, content_type=content_type)
 
+    @action(detail=False, url_path="import", methods=["post"])
+    def create_import(self, request: Request, **_kwargs) -> HttpResponse:
+        """
+        Create a new taxonomy and imports the tags from the uploaded file.
+        """
+        perm = "oel_tagging.import_taxonomy"
+        if not request.user.has_perm(perm):
+            raise PermissionDenied("You do not have permission to import taxonomies")
+
+        body = TaxonomyImportNewBodySerializer(data=request.data)
+        body.is_valid(raise_exception=True)
+
+        taxonomy_name = body.validated_data["taxonomy_name"]
+        taxonomy_description = body.validated_data["taxonomy_description"]
+        file = body.validated_data["file"].file
+        parser_format = body.validated_data["parser_format"]
+
+        taxonomy = create_taxonomy(taxonomy_name, taxonomy_description)
+        import_success = import_tags(taxonomy, file, parser_format)
+
+        if import_success:
+            return HttpResponse(status=200)
+        else:
+            import_error = "Error importing taxonomy"  # ToDo: Get actual error message
+            taxonomy.delete()
+            return HttpResponseBadRequest(import_error)
+
+    @action(detail=True, url_path="tags/import", methods=["put"])
+    def update_import(self, request: Request, **_kwargs) -> HttpResponse:
+        """
+        Imports tags from the uploaded file to an already created taxonomy.
+        """
+        perm = "oel_tagging.import_taxonomy"
+        if not request.user.has_perm(perm):
+            raise PermissionDenied("You do not have permission to import taxonomies")
```

Comment:
Same here. If a user can update a taxonomy, he can import tags to this taxonomy.

Reply:
Agreed.